R/robustbayesianmetaanalysis.R

```diff
@@ -0,0 +1,2419 @@
+          temp_row$rhat  <- results_table$`Rhat`[i]
+        }
+        
+        jasp_table$addRows(temp_row)
+      }
+    }
+    
+    # add footnote
+    jasp_table$addFootnote(
+      symbol = "\u002A",
+      paste0(
+        gettext("The weights"),
+        " (\u03C9) ",
+        gettext("correspond to"),
+        " ",
+        add_info$weight_type,
+        " ",
+        gettext("<em>p</em>-values.")
+      )
+    )
+    
+    return(jasp_table)
+  }
+.RoBMA_table_fill_studies   <-
+  function(jasp_table,
+           results_table,
+           add_info,
+           options,
+           individual = FALSE) {
+    # add columns
+    jasp_table$addColumnInfo(name = "terms",
+                             title = "",
+                             type = "string")
+    jasp_table$addColumnInfo(name = "mean",
+                             title = gettext("Mean"),
+                             type = "number")
+    jasp_table$addColumnInfo(name = "median",
+                             title = gettext("Median"),
+                             type = "number")
+    jasp_table$addColumnInfo(
+      name = "lowerCI",
+      title = gettext("Lower"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$results_CI)
+    )
+    jasp_table$addColumnInfo(
+      name = "upperCI",
+      title = gettext("Upper"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$results_CI)
+    )
+    if (individual) {
+      jasp_table$addColumnInfo(name = "error",
+                               title = gettext("error %"),
+                               type = "number")
+      jasp_table$addColumnInfo(name = "ess",
+                               title = gettext("ESS"),
+                               type = "integer")
+      jasp_table$addColumnInfo(name = "rhat",
+                               title = gettext("Rhat"),
+                               type = "number")
+    }
+    
+    # fill rows
+    if (!is.null(results_table)) {
+      temp_i <- 0
+      for (i in c(1:nrow(results_table))[grepl("theta", rownames(results_table))]) {
+        temp_i <- temp_i + 1
+        temp_row <- list(
+          terms    = add_info$study_names[temp_i],
+          mean     = results_table$Mean[i],
+          median   = results_table$Median[i],
+          lowerCI  = results_table[i, 3],
+          upperCI  = results_table[i, 4]
+        )
+        if (individual) {
+          temp_row$error <- results_table$`Error % of SD`[i]
+          temp_row$ess   <- results_table$`ESS`[i]
+          temp_row$rhat  <- results_table$`Rhat`[i]
+        }
+        
+        jasp_table$addRows(temp_row)
+      }
+    }
+    
+    # add footnote
+    if (add_info$effect_size == "r")
+      jasp_table$addFootnote(paste0(
+        gettext("Estimated studies' effects"),
+        " (\u03B8) ",
+        gettext("correspond to effect size"),
+        "\u03C1."
+      ))
+    if (add_info$effect_size == "d")
```

Comment:
if `if (add_info$effect_size == "d")` and `if (add_info$effect_size == "r")` a couple lines above are mutually exclusive then this should be an `else if`.

Reply:
yes, you are right. changed